resizing_hashtable/r_hashtables.py

```diff
@@ -51,15 +85,30 @@ def hash_table_remove(hash_table, key):
 # Should return None if the key is not found.
 # '''
 def hash_table_retrieve(hash_table, key):
-    pass
+    bucket_index = hash(key, hash_table.capacity)
+
+    existing_pair = hash_table.storage[bucket_index]
+    if existing_pair:
+        while existing_pair:
+            if existing_pair.key == key:
+                return existing_pair.value
+            existing_pair = existing_pair.next
+    return None
 
 
 # '''
 # Fill this in
 # '''
 def hash_table_resize(hash_table):
-    pass
+    resized = hash_table.capacity * 2
+    new_hash_table = HashTable(resized)
 
+    for i in hash_table.storage:
+        current = hash_table.storage[i]
+        while current is True:
+            hash_table_insert(new_hash_table, hash_table.storage[i].key, hash_table.storage[i].value)
```

Comment:
re-iterating from 1-1:
Note that hash_table.storage[i] will always be the first node in the linked list, causing line 109 to be problematic in repeatedly adding the same key/value to the new storage system.  with the current set up use of the 'current' variable, current.key and current.value will do what we are looking for.